stacks-signer/src/client/stackerdb.rs

```diff
@@ -160,28 +167,22 @@ impl<M: MessageSlotID + 'static> StackerDB<M> {
                 code: None,
             });
         };
+        let signer_pk = StacksPublicKey::from_private(&self.stacks_private_key);
         loop {
-            let mut slot_version = if let Some(versions) = self.slot_versions.get_mut(msg_id) {
-                if let Some(version) = versions.get(slot_id) {
-                    *version
-                } else {
-                    versions.insert(*slot_id, 0);
-                    1
-                }
-            } else {
-                let mut versions = HashMap::new();
-                versions.insert(*slot_id, 0);
-                self.slot_versions.insert(*msg_id, versions);
-                1
-            };
-
-            let mut chunk = StackerDBChunkData::new(slot_id.0, slot_version, message_bytes.clone());
-            chunk.sign(&self.stacks_private_key)?;
+            let slot_version = self
+                .signer_db
+                .get_latest_chunk_version(&signer_pk, slot_id.0)
+                .map_err(ClientError::SignerDBError)?
```

Comment:
nit: `thiserror` is handy
```suggestion
                .get_latest_chunk_version(&signer_pk, slot_id.0)?
```